api/serializers.py

```diff
@@ -1,12 +1,26 @@
 from django.contrib.auth.models import User
 from django.core.files.base import ContentFile
+from django.db.models.query_utils import Q
 from django.utils.timezone import now
 from rest_framework import serializers
 
 from .models import Profile, Report, Video
 
 
 class ProfileSerializer(serializers.ModelSerializer):
+    def to_representation(self, instance):
+        rep = super().to_representation(instance)
+        # get video
+        video = None
+        try:
+            video = Video.objects.get(
+                Q(sender=instance.user) & Q(receiver=instance.user)
+            )
+            video = read_video(video.video_file)
+        finally:
+            rep["video"] = video
+            return rep

```

Comment:
does this need an `except` ?

Reply:
All I would do in the `except` is ensure the value of `video` is none. Didn't seem worth it, since `video` is initialized to none

Reply:
I don't really get the try-finally here. Are you anticipating an exception in that code?

Reply:
Yes, the get will fail if there is no video